olp-cpp-sdk-dataservice-read/src/StreamLayerClientImpl.cpp

```diff
@@ -280,6 +280,101 @@ client::CancellableFuture<DataResponse> StreamLayerClientImpl::GetData(
 }
 
+client::CancellationToken StreamLayerClientImpl::Poll(
+    PollResponseCallback callback) {
+  auto poll_task = [=](client::CancellationContext context) -> PollResponse {
+    std::string subscription_id;
+    std::string subscription_mode;
+    std::string x_correlation_id;
+    std::shared_ptr<client::OlpClient> client;
+
+    {
+      std::lock_guard<std::mutex> lock(mutex_);
+      if (!client_context_) {
+        OLP_SDK_LOG_WARNING_F(kLogTag,
+                              "Poll: unsuccessful, not subscribed");
+
+        return client::ApiError(client::ErrorCode::PreconditionFailed,
+                                "Not subscribed", false);
+      }
+
+      subscription_id = client_context_->subscription_id;
+      subscription_mode = client_context_->subscription_mode;
+      x_correlation_id = client_context_->x_correlation_id;
+      client = client_context_->client;
+    }
+
+    OLP_SDK_LOG_INFO_F(kLogTag,
+                       "Poll: started, subscription_id=%s, "
+                       "subscription_mode=%s, x_correlation_id=%s",
+                       subscription_id.c_str(), subscription_mode.c_str(),
+                       x_correlation_id.c_str());
+
+    auto data =
+        StreamApi::ConsumeData(*client, layer_id_, subscription_id,
+                               subscription_mode, context, x_correlation_id);
+
+    if (!data.IsSuccessful()) {
+      OLP_SDK_LOG_WARNING_F(kLogTag, "Poll: consume data unsuccessful, error=%s",
+                            data.GetError().GetMessage().c_str());
+      return data.GetError();
+    }
+
+    auto res = data.MoveResult();
+
+    const auto& messages = res.GetMessages();
+
+    if (!messages.empty()) {
+      // Get offsets for all partitions presented in messages.
+      std::vector<model::StreamOffset> offsets;
+      for (const auto& message : messages) {
+        const auto current_offset = message.GetOffset();
+        auto iter = std::find_if(
+            offsets.begin(), offsets.end(),
+            [current_offset](const model::StreamOffset& offset) {
+              return offset.GetPartition() == current_offset.GetPartition();
+            });
+
+        if (iter != offsets.end()) {
+          *iter = current_offset;
+        } else {
+          offsets.push_back(current_offset);
+        }
+      }
+
+      // Commit offsets
+      model::StreamOffsets offsetsRequest;
+      offsetsRequest.SetOffsets(std::move(offsets));
+      auto commit_res = StreamApi::CommitOffsets(
+          *client, layer_id_, offsetsRequest, subscription_id,
+          subscription_mode, context, x_correlation_id);
+
+      if (!commit_res.IsSuccessful()) {
+        OLP_SDK_LOG_WARNING_F(kLogTag, "Poll: commit offsets unsuccessful, error=%s",
+                              commit_res.GetError().GetMessage().c_str());
+        return commit_res.GetError();
+      }
+
+      OLP_SDK_LOG_INFO_F(kLogTag, "Poll: done, response is successful.");
```

Comment:
```suggestion
      OLP_SDK_LOG_TRACE_F(kLogTag, "Poll: done, response is successful.");
```
The info level is a little too much, isn't it? Do we wish to be informed about every finished poll request?